docker_config_generator/templates/golang

```diff
@@ -0,0 +1,26 @@
+FROM golang:1.20-alpine AS builder
+
+RUN apk update && apk --no-cache upgrade
+RUN apk --no-cache add ca-certificates git
+WORKDIR /build
+# Fetch dependencies
+COPY go.mod go.sum ./
+RUN go mod download
+
+COPY . .
+RUN CGO_ENABLED=0 ${BUILD_COMMAND} .
```

Comment:
We can pass `CGO_ENABLED` 's value as a Build Arg and add it to the template , So that user can switch if required

Reply:
The `BUILD_COMMAND` arg is not present in Dockerfile ig. Need to add that

Reply:
I think `CGO_ENABLED` should be made 0 compulsorily since it means static build without dynamic linking, and no problem would arise while running ig